YAFC/Workspace/ProductionTable/ProductionTableFlatHierarchy.cs

```diff
@@ -187,5 +234,25 @@ public class FlatHierarchy<TRow, TGroup> where TRow : ModelObject<TGroup>, IGrou
         public void BuildHeader(ImGui gui) {
             grid.BuildHeader(gui);
         }
+
+        private static SchemeColor GetHighlightingBackgroundColor(RowHighlighting highlighting) {
+            return highlighting switch {
+                RowHighlighting.Green => SchemeColor.TagColorGreenBackground,
+                RowHighlighting.Yellow => SchemeColor.TagColorYellowBackground,
+                RowHighlighting.Red => SchemeColor.TagColorRedBackground,
+                RowHighlighting.Blue => SchemeColor.TagColorBlueBackground,
+                _ => SchemeColor.None
+            };
+        }
+
+        private static SchemeColor GetHighlightingTextColor(RowHighlighting highlighting) {
```

Comment:
So this (finding the text color for a specific background color) is (should be) automatic by the GUI/coloring system.

I do not know (from the top of my head) how it exactly works though. If you want/need I can take a look in the (GUI) code...

Reply:
This could be solved if I add the color quadruples in the correct way (as you pointed out). On my todo list.

Reply:
The colors are there now. But how would I use the coloring system? Could you provide an explanation or outline some of the key points?

Reply:
I searched a bit, but unfortunately it is not centered in a single (helper) class/function... But there are lots of places where the (G)UI assumes the order of colors in the quadruple:
* [find disabled text color](https://github.com/have-fun-was-taken/yafc-ce/blob/aa077c816c9ea80dec4dff612bf1855b54a339eb/YAFC/Widgets/ImmediateWidgets.cs#L96)
* [Finding hover color](https://github.com/have-fun-was-taken/yafc-ce/blob/aa077c816c9ea80dec4dff612bf1855b54a339eb/YAFC/Widgets/ImmediateWidgets.cs#L60)

When you (re)search (in) the code, you'll notice all kinds of these assumptions. Which would be the reason to keep the order/quadruples in the ColorSchme tags.

It would be much nicer the quadruples are stored in/as their own class, with helper functions to find the correct color like `getHoverColor()`, `getTextColor()`, etc... We should include this improvement, when we take a good look at improving the color/theme functionality.

Reply:
Yes, that would be a good idea. See #104 😉 

For the time being, do you think this conversation is resolved? I'm not sure if I can change this in a reasonable amount of time.